src/huntsman/pocs/observatory.py

```diff
@@ -189,165 +190,87 @@ def take_flat_fields(self, camera_names=None, alt=None, az=None, safety_func=Non
-
-            # Create dark observation
-            dark_obs = self._create_dark_observation(exptime)
-
-            # Loop over exposure times for each camera.
-            for num in range(num_darks):
-
-                self.logger.debug(f'Taking dark {num + 1} of {num_darks} with exptime={exptime} s.')
-
-                camera_events = dict()
-
-                # Take a given number of exposures for each exposure time.
-                for camera in cameras_list.values():
-
-                    # Set header
-                    fits_headers = self.get_standard_headers(observation=dark_obs)
-                    # Common start time for cameras
-                    fits_headers['start_time'] = flatten_time(start_time)
-
-                    # Create filename
-                    path = os.path.join(dark_obs.directory, camera.uid, dark_obs.seq_time)
-
-                    filename = os.path.join(path, f'{imtype}_{num:02d}.{camera.file_extension}')
-
-                    # Take picture and get event
-                    camera_event = camera.take_observation(
-                        dark_obs,
-                        fits_headers,
-                        filename=filename,
-                        exptime=exptime,
-                        dark=True,
-                        blocking=False
-                    )
-
-                    self.logger.debug(f'Camera {camera.uid} is exposing for {exptime}s')
-
-                    camera_events[camera] = {
-                        'event': camera_event,
-                        'filename': filename,
-                    }
-
-                    darks_filenames.append(filename)
-
-                    self.logger.debug(camera_events)
-
-                # Block until done exposing on all cameras
-                self.logger.debug('Waiting for dark images...')
-                wait_for_events(list(info['event'] for info in camera_events.values()))
-        self.logger.debug(darks_filenames)
-        return darks_filenames
+        if cameras is None:
+            cameras = self.cameras
+        # Create the observation
+        position = self.mount.get_current_coordinates()
+        observation = BiasObservation(position=position)
+        # Take the observation (blocking)
+        self._take_observation(observation, cameras=cameras, **kwargs)
+
+    def take_dark_observation(self, exptimes=None, cameras=None, **kwargs):
+        """ Take a dark observation block on each camera (blocking).
+        Args:
+            cameras (dict, optional): Dict of cam_name: camera pairs. If None (default), use all
+                the cameras.
+        """
+        if cameras is None:
+            cameras = self.cameras
+        # Create the observation
+        position = self.mount.get_current_coordinates()
+        observation = DarkObservation(exptimes=exptimes, position=position)
+        # Take the observation (blocking)
+        self._take_observation(observation, cameras=cameras, **kwargs)
```

Comment:
Same here, isn't it `_take_observation_block` instead of `_take_observation`? Cause that's what the docstrings say